libraries/core-react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,235 @@
+import React, { useState, useCallback, forwardRef } from 'react'
+import DatePicker, {
+  ReactDatePickerProps,
+  registerLocale,
+} from 'react-datepicker'
+import 'react-datepicker/dist/react-datepicker.css'
+import enGb from 'date-fns/locale/en-GB'
+import styled from 'styled-components'
+import { Icon } from '../Icon'
+import { calendar } from '@equinor/eds-icons'
+import { PopupHeader } from './PopupHeader'
+import { datePicker as tokens, DatePickerToken } from './DatePicker.tokens'
+
+registerLocale('en-gb', enGb)
+
+export interface DatePickerProps {
+  id: string
+  value: Date | undefined | null
+  label: string
+  onChanged?: (date: Date | null) => void
+  disableFuture?: boolean
+  disableBeforeDate?: Date
+  className?: string
+  dateFormat?: string
+  placeholder?: string
+  readOnly?: boolean
+  popperPlacement?:
+    | 'auto-start'
+    | 'auto'
+    | 'auto-end'
+    | 'top-start'
+    | 'top'
+    | 'top-end'
+    | 'right-start'
+    | 'right'
+    | 'right-end'
+    | 'bottom-end'
+    | 'bottom'
+    | 'bottom-start'
+    | 'left-end'
+    | 'left'
+    | 'left-start'
+  locale?: string
+}
```

Comment:
Missing extension of React native attributes for a component. Since the "main" component here is the input field this would be: `InputHTMLAttributes<HTMLInputElement>`